bedrock/src/main/java/no/apps/bedrock/ui/controller/AppsController.kt

```diff
@@ -109,28 +111,30 @@ abstract class AppsController<B : ViewBinding, A : PageArgs, VMA : Any, VM : App
     }
 
     @CallSuper
-    protected open fun navigate(args: PageArgs) {
-        if (pendingNavigation || !isAttached) {
-            Timber.w("Skipping navigation: pending: $pendingNavigation, attached: $isAttached")
-            return
-        }
-        pendingNavigation = true
-        launch {
-            withTimeoutOrNull(MAX_NAVIGATION_BLOCK) {
-                suspendCoroutine<Unit> {
-                    detachContinuation = it
+    protected open fun navigate(args: PageArgs) =
+        closeKeyboardThen {
+            if (pendingNavigation || !isAttached) {
+                Timber.w("Skipping navigation: pending: $pendingNavigation, attached: $isAttached")
+                return@closeKeyboardThen
+            }
+            pendingNavigation = true
+            launch {
+                withTimeoutOrNull(MAX_NAVIGATION_BLOCK) {
+                    suspendCoroutine<Unit> {
+                        detachContinuation = it
+                    }
                 }
+                detachContinuation = null
+                pendingNavigation = false
             }
-            detachContinuation = null
-            pendingNavigation = false
+            navigator.navigate(pageArgs, args, this)
         }
-        navigator.navigate(pageArgs, args, this)
-    }
 
     @CallSuper
-    protected open fun goBack() {
-        activity?.onBackPressed()
-    }
+    protected open fun goBack() =
```

Comment:
I suggest to add a parameter with default value that can control if the keyboard should be hidden or not before.